mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -769,10 +769,24 @@ pub enum NewClientError {
     InvalidEndpoint(#[from] EndpointError),
     /// Invalid AWS credentials
     #[error("invalid AWS credentials")]
-    ProviderFailure(#[from] mountpoint_s3_crt::common::error::Error),
+    ProviderFailure(#[source] mountpoint_s3_crt::common::error::Error),
     /// Invalid configuration
     #[error("invalid configuration: {0}")]
     InvalidConfiguration(String),
+    /// An internal error from within the AWS Common Runtime
+    #[error("Unknown CRT error")]
+    CrtError(#[source] mountpoint_s3_crt::common::error::Error),
+}
+
+impl NewClientError {
+    fn construction_error(e: mountpoint_s3_crt::common::error::Error) -> Self {
+        const AWS_ERROR_INVALID_ARGUMENT: i32 = 34;
+        if e.raw_error() == AWS_ERROR_INVALID_ARGUMENT {
+            Self::InvalidConfiguration("see crt logs for details".into())
```

Comment:
Would using `e.to_string()` gives a bit better hint for the error?

Reply:
I've removed this method and just wrapped the CRT error for now. The error message may be even less informative, but we can fix that at a later time.